adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -321,6 +321,9 @@ public AuthenticationResult getAuthTokenInBackground(final AuthenticationRequest
 
         verifyNotOnMainThread();
 
+        // populate BrokerEvent info re installed Brokers
+        setAndLogBrokerInstallationStatuses(brokerEvent);
```

Comment:
Is there any reason that we put the installation check here instead of canSwitchToBroker or other upriver functions before talking to broker?

Reply:
`canSwitchToBroker` would be a better place for the logging, I agree. I'll make this change to relocate logging there -- because I need access to a `BrokerEvent` object to report this status in Telemetry, I can't put the telemetry code there. A better alternative isn't immediately clear to me, but I can look around while I'm in the code 👍

Reply:
Logging has been moved to `BrokerProxy#canSwitchToBroker` in ce5ccaf4404503a6f79e229af891743b539e14c0

`BrokerEvent` fields are still set in the same place, as `BrokerProxy#getIntentForBrokerActivity` and `BrokerProxy#getAuthTokenInBackground` appear to be the only places where we have access both to the `BrokerEvent` object and a `Context` object (to acquire the `PackageManager`)